scene/gui/tree.cpp

```diff
@@ -4215,10 +4215,22 @@ Tree::SelectMode Tree::get_select_mode() const {
 }
 
 void Tree::deselect_all() {
+	if (root) {
+		for (int i = 0; i < columns.size(); ++i) {
+			if (root->is_selected(i)) {
```

Comment:
Is the check meaningful at this point? We are already iterating over all cells. And if it is meaningful, why don't we do the same for other items we iterate below?

IMO if calling `deselect()` on an unselected column is slow, then we should optimize that method instead. I can see that neither `TreeItem::deselect` nor `Tree::item_deselected` perform any checks for the current state. So I'd improve that instead, if we really ought to.